cpu/native/netdev_raw802154/netdev_raw802154.c

```diff
@@ -0,0 +1,337 @@
+ * more details.
+ */
+
+#include <assert.h>
+#include <err.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+/* needs to be included before native's declarations of ntohl etc. */
+#include "byteorder.h"
+
+#ifdef __MACH__
+#define _POSIX_C_SOURCE
+#include <net/if.h>
+#undef _POSIX_C_SOURCE
+#include <sys/types.h>
+#include <ifaddrs.h>
+#include <net/if_dl.h>
+#elif defined(__FreeBSD__)
+#include <sys/socket.h>
+#include <net/if.h>
+#include <ifaddrs.h>
+#include <net/if_dl.h>
+#else
+#include <net/if.h>
+#include <linux/if_tun.h>
+#include <linux/if_ether.h>
+#include <linux/if_packet.h>
+#endif
+
+#include "native_internal.h"
+
+#include "net/netdev.h"
+#include "netdev_raw802154.h"
+#include "net/netopt.h"
+
+#include "periph/hwrng.h"
+
+#include "async_read.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef ETH_P_IEEE802154
+#define ETH_P_IEEE802154 0x00F6
+#endif
+
+#define RAW802154_DEFAULT_ADDR_PAN      (0x0023)
+#define RAW802154_DEFAULT_ADDR_SHORT    (0xfffe)
+#define RAW802154_DEFAULT_ADDR_LONG     (0x1222334455667788)
+
+/* support one raw802154 interface for now */
+netdev_raw802154_t netdev_raw802154;
+
+void _set_page(netdev_raw802154_t *dev, uint8_t page)
+{
+    dev->page = page;
+}
+
+static uint8_t _get_page(netdev_raw802154_t *dev)
+{
+    return dev->page;
+}
+
+static void _set_chan(netdev_raw802154_t *dev, uint8_t channel)
+{
+    dev->netdev.chan = channel;
+}
+
+static inline void _isr(netdev_t *netdev)
+{
+    if (netdev->event_callback)
+        netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
+}
+
+static void _raw802154_isr(int fd, void *arg) {
+    (void) fd;
+    (void) arg;
+    netdev_t *netdev = (netdev_t *)&netdev_raw802154;
+
+    if (netdev->event_callback)
+        netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+}
+
+static void _set_addr_short(netdev_raw802154_t *dev, uint16_t addr)
+{
+    dev->netdev.short_addr[0] = (uint8_t)(addr);
+    dev->netdev.short_addr[1] = (uint8_t)(addr >> 8);
+}
+
+static void _set_addr_long(netdev_raw802154_t *dev, uint64_t addr)
+{
+    for (int i = 0; i < 8; i++)
+        dev->netdev.long_addr[i] = (uint8_t)(addr >> (i * 8));
+}
+
+static void _set_pan(netdev_raw802154_t *dev, uint16_t pan)
+{
+    dev->netdev.pan = pan;
+}
+
+static int16_t _get_txpower(netdev_raw802154_t *dev)
+{
+    (void)dev;
+    return 0;
+}
+
+static uint8_t _get_max_retries(netdev_raw802154_t *dev)
+{
+    (void)dev;
+    return 0;
+}
+
+#define _MAX_MHR_OVERHEAD   (25)
+static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+
+    switch (opt) {
+        case NETOPT_STATE:
+            if (max_len < sizeof(netopt_state_t)) {
+                return -EOVERFLOW;
+            }
```

Comment:
We used to do it like that, but please use (`assert()`) now instead (and fail if the type is `!=` for scalar, and `<` only for array types). Same of course fall all options below.